packages/devtools_app/lib/src/screens/network/network_controller.dart

```diff
@@ -388,6 +390,25 @@ class NetworkController extends DisposableController
     }
   }
 
+  @override
+  OfflineScreenData prepareOfflineScreenData() {
+    final requests =
+        filteredData.value.whereType<DartIOHttpRequestData>().toList();
+    return OfflineScreenData(
+      screenId: NetworkScreen.id,
+      data: serializeRequestDataForOfflineMode(requests),
```

Comment:
I recommend creating a class `OfflineNetworkData with Serializable` that packages up all of the data stored on the Network  controller for serialization. This includes the list of requests as well as other information like the currently selected request. In the future we may want to include the current filter, but let's leave that out for now. 

Creating a new class `OfflineNetworkData` will make it easier to test the `toJson` and `fromJson` methods that should be implemented on the class. See other examples of classes that mixin `Serializable` in DevTools for an example.

Then here, you'd create an instance of `OfflineNetworkData` from the current data on the network controller, and call `toJson()` to get the value for the `data` parameter of `OfflineScreenData` in this return statement.

Reply:
Yup, sure!

Reply:
done